SpiceSharp/Simulations/Base/BaseSimulation.cs

```diff
@@ -0,0 +1,591 @@
+                    Statistics.DecompositionTime.Stop();
+                }
+
+                // Solve the equation
+                Statistics.SolveTime.Start();
+                matrix.Solve(state.Rhs, state.Rhs);
+                Statistics.SolveTime.Stop();
+
+                // The result is now stored in the RHS vector, let's move it to the current solution vector
+                state.StoreSolution();
+
+                // Reset ground nodes
+                state.Rhs[0] = 0.0;
+                state.Solution[0] = 0.0;
+                state.OldSolution[0] = 0.0;
+
+                // Exceeded maximum number of iterations
+                if (iterno > maxIterations)
+                {
+                    Statistics.Iterations += iterno;
+                    return false;
+                }
+
+                if (state.IsConvergent && iterno != 1)
+                    state.IsConvergent = IsConvergent();
+                else
+                    state.IsConvergent = false;
+
+                switch (state.Init)
+                {
+                    case RealState.InitializationStates.InitFloat:
+                        if (state.UseDC && state.HadNodeSet)
+                        {
+                            if (pass)
+                                state.IsConvergent = false;
+                            pass = false;
+                        }
+                        if (state.IsConvergent)
+                        {
+                            Statistics.Iterations += iterno;
+                            return true;
+                        }
+                        break;
+
+                    case RealState.InitializationStates.InitJunction:
+                        state.Init = RealState.InitializationStates.InitFix;
+                        state.Sparse |= RealState.SparseStates.ShouldReorder;
+                        break;
+
+                    case RealState.InitializationStates.InitFix:
+                        if (state.IsConvergent)
+                            state.Init = RealState.InitializationStates.InitFloat;
+                        pass = true;
+                        break;
+
+                    case RealState.InitializationStates.InitTransient:
+                        if (iterno <= 1)
+                            state.Sparse = RealState.SparseStates.ShouldReorder;
+                        state.Init = RealState.InitializationStates.InitFloat;
+                        break;
+
+                    case RealState.InitializationStates.Init:
+                        state.Init = RealState.InitializationStates.InitFloat;
+                        break;
+
+                    default:
+                        Statistics.Iterations += iterno;
+                        throw new CircuitException("Could not find flag");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Load the circuit with the load behaviors
+        /// </summary>
+        protected void Load()
```

Comment:
@svenboulanger Consider making it shorter

Reply:
We may review this at a later date.

Reply:
I have turned it into an issue #24